molecule-runtime/src/commonMain/kotlin/app/cash/molecule/molecule.kt

```diff
@@ -130,8 +177,8 @@ public fun <T> CoroutineScope.launchMolecule(
  */
 public fun <T> CoroutineScope.launchMolecule(
   mode: RecompositionMode,
-  emitter: (value: T) -> Unit,
-  body: @Composable () -> T,
+  emitter: Emitter<T>,
+  body: @Composable (Emitter<T>) -> Unit,
```

Comment:
This would be a binary breaking change. In all likelihood this function would stay as-is and a new one would have to be added with a detailed description of when and why you would choose on vs. the other.